src/main/java/gobblin/pnda/PNDARegistryBasedConverter.java

```diff
@@ -57,20 +57,21 @@ public void close() throws IOException {
     delegate.close();
   }
 
-  @Override
+
+@Override
   public Schema convertSchema(String inputSchema, WorkUnitState workUnit) throws SchemaConversionException {
 
     TopicConfig config = PNDARegistry.getConfig(inputSchema);
     if (config == TopicConfig.NOCONFIG) {
         config = GobblinConfigRegistry.getConfig(inputSchema, workUnit);
     }
 
-    String className = (config == null) ? null : config.getConverterClass();
+    String className = config.getConverterClass();
     if (null != className) {
       try {
         Class<?> clazz = Class.forName(className);
         Constructor<?> ctor = clazz.getConstructor();
-        delegate = (PNDAAbstractConverter<?, TopicConfig>) ctor.newInstance();
+        delegate = newInstance(ctor);
```

Comment:
Can we rather inline the suppress statement?
This can be done using a local variable: https://docs.oracle.com/javase/6/docs/api/java/lang/annotation/ElementType.html#LOCAL_VARIABLE
Something like this:
@SuppressWarnings("unchecked")
PNDAAbstractConverter<?, TopicConfig> converter = (PNDAAbstractConverter<?, TopicConfig>) ctor.newInstance();
delegate = converter;

Reply:
Done